pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -88,6 +89,49 @@ func (k *Kaniko) BuildAndPushContainerImage(buildOptions *BuildOptions, namespac
 	return errors.New("Kaniko job has timed out")
 }
 
+func (k *Kaniko) GetOnbuildStages(onbuildArtifacts []runtime.Artifact) ([]string, error) {
```

Comment:
Wouldn't it be nicer to returned named stages? That way you don't have to do the whole artifact index thing and it'll be easier to read